core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,74 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.slf4j.{Marker, MarkerFactory}
 
-trait Logging {
-  val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
 
-  protected var logIdent: String = null
+object Log4jControllerRegistration {
+  private val logger = Logger(this.getClass.getName)
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.getDeclaredConstructor().newInstance()
+    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+    logger.info("Registered kafka:type=kafka.Log4jController MBean")
+  } catch {
+    case _: Exception => logger.info("Couldn't register kafka:type=kafka.Log4jController MBean")
+  }
+}
+
+trait Logging extends LazyLogging {
+  def loggerName: String = logger.underlying.getName
+
+  protected var logIdent: String = _
+
+  Log4jControllerRegistration
 
-  protected def msgWithLogIdent(msg: String) =
+  protected def msgWithLogIdent(msg: String): String =
     if (logIdent == null) msg else logIdent + msg
 
-  def trace(msg: => String): Unit = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg))
-  }
-  def trace(e: => Throwable): Any = {
-    if (logger.isTraceEnabled())
-      logger.trace(logIdent,e)
-  }
-  def trace(msg: => String, e: => Throwable) = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg),e)
-  }
-  def swallowTrace(action: => Unit) {
-    CoreUtils.swallow(logger.trace, action)
-  }
+  val FatalMarker: Marker = MarkerFactory.getMarker("FATAL")
```

Comment:
I suggested this should be in the companion object of `Logging`, not the trait.

Reply:
Sorry I think I misunderstood you. Just to confirm, are you looking for something like this?
```
private object Logging {
  val FatalMarker: Marker = MarkerFactory.getMarker("FATAL")
}
trait Logging extends LazyLogging {
  // ...
  def fatal(msg: => String): Unit =
    logger.error(Logging.FatalMarker, msgWithLogIdent(msg))
}

Reply:
Yes.